### PR TITLE
Use notistack for Errors

### PR DESCRIPTION
**Pull request by jessie-ross**
This PR implements notistack for non-fatal user error messages:
- Saves on code relating to snackbars/alerts
- Provides message variants like 'error'
- Provides a better interface for code-reuse
- Decouples errors from the "layout"

Notistack maintainability case:
- Minimal, single purpose
- Small surface area `<SnackbarProvider>`, `useSnackbar`, `enqueueSnackbar`, `closeSnackbar`
- Two years old, (young but not too young)
- Active maintainer
- Recommended by MUI (a good sign).

Resolves: #183
Replaces: #193 
Fixe half of the problem of: #200 

## General thoughts

Something to think of is if we want to elevate more errors to fatal level.
In fact it looks like we don't have fatal errors: disabling any interaction and only showing a message telling a user to refresh with a refresh button as well. 

This is a good option if we want to ensure valid states on the front-end but comes at the cost of UX.

## How has this been tested?
- AT - Getting the hang of in-depth testing
- MT

### CODE REVIEW

**Comment by jessie-ross** [comment]
Ah I need to remove the existing system, will push that shortly.

**Comment by jessie-ross** [comment]
And oops I was wrong about not having fatal errors

File: utils/data-loading.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export function useDataLoadingError<E extends Error | null>(error: E, dataName?: string) {
+  const { enqueueSnackbar } = useSnackbar()
+
+  useEffect(() => {
+    if (error) {
+      const userErrorMessage = dataName
+        ? `Oops! There was a problem loading some data of type: ${dataName}.`
+        : 'Oops! There was a problem loading some data.'
+      enqueueSnackbar(userErrorMessage, { variant: 'error' })
```
Comment (yanirs) [inline utils/data-loading.ts:12]: What do you think about using `persist: true`? I played with it a bit, and it feels like the errors disappear too quickly.

File: utils/data-loading.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { useSnackbar } from 'notistack'
+import { useEffect } from 'react'
+
+export function useDataLoadingError<E extends Error | null>(error: E, dataName?: string) {
```
Comment (yanirs) [inline utils/data-loading.ts:4]: Can you please add a short docstring?

Also, it looks like `dataName` isn't being used. If it's not useful, let's remove it for now. If it is useful, let's use it. :slightly_smiling_face:

Comment (jessie-ross) [reply utils/data-loading.ts:4]: I plan to use it very soon with #208

File: utils/data-loading.ts (hunk 1 above)
Comment (jessie-ross) [reply utils/data-loading.ts:12]: Yeah I am ok with that.

**Comment by jessie-ross** [comment]
Ok made it persist, should be good to go.

**Pull request merged**